src/sentry/migrations/0637_remove_pr_comment_pr_foreignkey_db_constraint.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 3.2.23 on 2024-01-24 17:07
+
+import django.db.models.deletion
+from django.db import migrations
+
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0636_monitor_incident_env_resolving_index"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="pullrequestcomment",
+            name="pull_request",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                db_constraint=False,
+                on_delete=django.db.models.deletion.CASCADE,
```

Comment:
The `on_delete` clause won't do anything without the constraint.

Reply:
so it won't cascade if the PR itself is deleted?

Reply:
does this mean i need to add some sort of custom deletion logic to delete `PullRequestComment` objects appropriately?

Reply:
on second glance, i was querying for the uniqueness constraints incorrectly. i fixed the query and found the offending constraint